flink-clients/src/main/java/org/apache/flink/client/cli/CliFrontendParser.java

```diff
@@ -459,4 +476,14 @@ public static InfoOptions parseInfoCommand(String[] args) throws CliArgsExceptio
 		}
 	}
 
+	public static MainOptions parseMainCommand(String[] args) throws CliArgsException {
+		try {
+			DefaultParser parser = new DefaultParser();
+			CommandLine line = parser.parse(MAIN_OPTIONS, args, false);
```

Comment:
You're parsing the entire arguments here. What if the user jar has `--configDir` as parameter?